--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2796,6 +_,7 @@
     }
 
     protected void updateSwingTime() {
+        if (!this.swinging && this.swingTime == 0) return; // Kobosh - Lithium - skip swing update when idle
         int currentSwingDuration = this.getCurrentSwingDuration();
         if (this.swinging) {
             this.swingTime++;
@@ -3798,6 +_,7 @@
     protected void updateFallFlying() {
         this.checkFallDistanceAccumulation();
         if (!this.level().isClientSide()) {
+            if (!this.isFallFlying() && this.fallFlyTicks == 0) return; // Kobosh - Lithium - skip gliding update when inactive
             if (!this.canGlide()) {
                 if (this.getSharedFlag(Entity.FLAG_FALL_FLYING) && !CraftEventFactory.callToggleGlideEvent(this, false).isCancelled()) // CraftBukkit
                 this.setSharedFlag(Entity.FLAG_FALL_FLYING, false);
