--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -815,11 +_,13 @@
             return;
         }
         // Paper end - Allow nerfed mobs to jump and float
+        int i = this.tickCount + this.getId(); // Kobosh - Petal - reduce sensor work
         ProfilerFiller profilerFiller = Profiler.get();
         profilerFiller.push("sensing");
-        this.sensing.tick();
+        if (i % me.earthme.luminol.config.modules.optimizations.PetalReduceSensorWorkConfig.delayTicks == 0 || !me.earthme.luminol.config.modules.optimizations.PetalReduceSensorWorkConfig.enabled) {
+            this.sensing.tick();
+        }
         profilerFiller.pop();
-        int i = this.tickCount + this.getId();
         if (i % 2 != 0 && this.tickCount > 1) {
             profilerFiller.push("targetSelector");
             this.targetSelector.tickRunningGoals(false);
