--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -835,11 +_,14 @@
             return;
         }
         // Paper end - Allow nerfed mobs to jump and float
+        int idBasedTickCount = this.tickCount + this.getId(); // Kobosh - Petal - reduce sensor work
         ProfilerFiller profiler = Profiler.get();
         profiler.push("sensing");
-        this.sensing.tick();
+        int sensorDelay = me.earthme.luminol.config.modules.optimizations.PetalReduceSensorWorkConfig.delayTicks; // Kobosh - Petal - guard against delayTicks <= 0
+        if (!me.earthme.luminol.config.modules.optimizations.PetalReduceSensorWorkConfig.enabled || sensorDelay <= 0 || idBasedTickCount % sensorDelay == 0) { // Kobosh - Petal - guard against delayTicks <= 0
+            this.sensing.tick();
+        }
         profiler.pop();
-        int idBasedTickCount = this.tickCount + this.getId();
         if (idBasedTickCount % 2 != 0 && this.tickCount > 1) {
             profiler.push("targetSelector");
             this.targetSelector.tickRunningGoals(false);
