--- a/net/minecraft/server/network/LegacyQueryHandler.java
+++ b/net/minecraft/server/network/LegacyQueryHandler.java
@@ -22,6 +_,11 @@
 
     @Override
     public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
+        // Kobosh - Krypton: discard immediately on inactive channels
+        if (!ctx.channel().isActive()) {
+            ((ByteBuf) msg).clear();
+            return;
+        }
         ByteBuf in = (ByteBuf)msg;
         // Paper start - Make legacy ping handler more reliable
         if (this.buf != null) {
