--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2853,7 +_,7 @@
         } // Folia end - rewrite login process - move connection ownership to global region
         this.waitingForSwitchToConfig = true; // Folia - rewrite login process - fix bad ordering of this field write - moved down
         this.send(ClientboundStartConfigurationPacket.INSTANCE);
-        this.connection.setupOutboundProtocol(ConfigurationProtocols.CLIENTBOUND);
+        this.connection.setupOutboundProtocolAsync(ConfigurationProtocols.CLIENTBOUND, null, true); // Kobosh - async protocol switching optimization
     }
 
     @Override
@@ -3718,12 +_,12 @@
             throw new IllegalStateException("Client acknowledged config, but none was requested");
         } else {
             final ServerConfigurationPacketListenerImpl listener = new ServerConfigurationPacketListenerImpl(this.server, this.connection, this.createCookie(this.player.clientInformation())); // Paper
-            this.connection
-                .setupInboundProtocol(
-                    ConfigurationProtocols.SERVERBOUND,
-                    listener // Paper
-                );
-            new io.papermc.paper.event.connection.configuration.PlayerConnectionReconfigureEvent(listener.paperConnection).callEvent(); // Paper
+            this.connection.setupInboundProtocolAsync(
+                ConfigurationProtocols.SERVERBOUND,
+                listener,
+                () -> new io.papermc.paper.event.connection.configuration.PlayerConnectionReconfigureEvent(listener.paperConnection).callEvent(),
+                true
+            ); // Kobosh - async protocol switching optimization
         }
     }
 
