From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: koboshchan <lucaszhang1001@gmail.com>
Date: Fri, 15 May 2026 23:09:08 -0700
Subject: [PATCH] Ender Chest: 6 rows by permission


diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index b39325b3c11b689a2fe7a57f679ec6a38563b257..dbd6c47a6d959b02a4c28a351a8d0c0a7251b8c5 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -179,6 +179,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     private int currentImpulseContextResetGraceTime = 0;
     public boolean affectsSpawning = true; // Paper - Affects Spawning API
     public net.kyori.adventure.util.TriState flyingFallDamage = net.kyori.adventure.util.TriState.NOT_SET; // Paper - flying fall damage
+    public int sixRowEnderchestSlotCount = -1; // kobosh - Barrels and enderchests 6 rows
 
     // CraftBukkit start
     public boolean fauxSleeping;
diff --git a/net/minecraft/world/inventory/ChestMenu.java b/net/minecraft/world/inventory/ChestMenu.java
index 1fb3a34ea872dec73658fac59743e46ef8db5151..da4e5f0ae8356a0f1bbfbaa2d67191971e2aa5cd 100644
--- a/net/minecraft/world/inventory/ChestMenu.java
+++ b/net/minecraft/world/inventory/ChestMenu.java
@@ -70,6 +70,24 @@ public class ChestMenu extends AbstractContainerMenu {
         return new ChestMenu(MenuType.GENERIC_9x3, containerId, playerInventory, container, 3);
     }
 
+    // Kobosh start - Barrels and enderchests 6 rows
+    public static ChestMenu oneRow(int syncId, Inventory playerInventory, Container inventory) {
+        return new ChestMenu(MenuType.GENERIC_9x1, syncId, playerInventory, inventory, 1);
+    }
+
+    public static ChestMenu twoRows(int syncId, Inventory playerInventory, Container inventory) {
+        return new ChestMenu(MenuType.GENERIC_9x2, syncId, playerInventory, inventory, 2);
+    }
+
+    public static ChestMenu fourRows(int syncId, Inventory playerInventory, Container inventory) {
+        return new ChestMenu(MenuType.GENERIC_9x4, syncId, playerInventory, inventory, 4);
+    }
+
+    public static ChestMenu fiveRows(int syncId, Inventory playerInventory, Container inventory) {
+        return new ChestMenu(MenuType.GENERIC_9x5, syncId, playerInventory, inventory, 5);
+    }
+    // Kobosh end - Barrels and enderchests 6 rows
+
     public static ChestMenu sixRows(int containerId, Inventory playerInventory, Container container) {
         return new ChestMenu(MenuType.GENERIC_9x6, containerId, playerInventory, container, 6);
     }
diff --git a/net/minecraft/world/inventory/PlayerEnderChestContainer.java b/net/minecraft/world/inventory/PlayerEnderChestContainer.java
index 9749909b8253b432bb2f7fba2cd8ff17a8579b30..aa8d960bbf0e99090ba37f273ee5ec8c0e9908cc 100644
--- a/net/minecraft/world/inventory/PlayerEnderChestContainer.java
+++ b/net/minecraft/world/inventory/PlayerEnderChestContainer.java
@@ -26,11 +26,18 @@ public class PlayerEnderChestContainer extends SimpleContainer {
     }
 
     public PlayerEnderChestContainer(Player owner) {
-        super(27);
+        super(54); // Kobosh - Barrels and enderchests 6 rows
         this.owner = owner;
         // CraftBukkit end
     }
 
+    // Kobosh start - Barrels and enderchests 6 rows
+    @Override
+    public int getContainerSize() {
+        return owner == null || owner.sixRowEnderchestSlotCount < 0 ? super.getContainerSize() : owner.sixRowEnderchestSlotCount;
+    }
+    // Kobosh end - Barrels and enderchests 6 rows
+
     public void setActiveChest(EnderChestBlockEntity enderChestBlockEntity) {
         this.activeChest = enderChestBlockEntity;
     }
diff --git a/net/minecraft/world/level/block/EnderChestBlock.java b/net/minecraft/world/level/block/EnderChestBlock.java
index d65a45fb9e012c0d9d0b599128675af75f4fc7ad..5091757860f20ee1ec6a90b3d79296878657dc8f 100644
--- a/net/minecraft/world/level/block/EnderChestBlock.java
+++ b/net/minecraft/world/level/block/EnderChestBlock.java
@@ -86,7 +86,7 @@ public class EnderChestBlock extends AbstractChestBlock<EnderChestBlockEntity> i
                 enderChestInventory.setActiveChest(enderChestBlockEntity); // Needs to happen before ChestMenu.threeRows as it is required for opening animations
                 if (level instanceof ServerLevel serverLevel && player.openMenu(
                     new SimpleMenuProvider(
-                        (containerId, playerInventory, player1) -> ChestMenu.threeRows(containerId, playerInventory, enderChestInventory), CONTAINER_TITLE
+                        (containerId, playerInventory, player1) -> getEnderChestSixRows(containerId, playerInventory, player, enderChestInventory), CONTAINER_TITLE // kobosh - Barrels and enderchests 6 rows
                     )
                 ).isPresent()) {
                 // Paper end - Fix InventoryOpenEvent cancellation - moved up;
