From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: koboshchan <lucaszhang1001@gmail.com>
Date: Tue, 19 May 2026 03:39:12 -0700
Subject: [PATCH] Tps Bar


diff --git a/src/main/java/org/bukkit/Chunk.java b/src/main/java/org/bukkit/Chunk.java
index d873b78f6935a23ff8f6092b80968267b182cc0f..17b914c2fdb5e40c45fc224721b4f4bfc4d038cf 100644
--- a/src/main/java/org/bukkit/Chunk.java
+++ b/src/main/java/org/bukkit/Chunk.java
@@ -392,4 +392,6 @@ public interface Chunk extends PersistentDataHolder {
          */
         UNLOADED;
     }
+
+    long getChunkHotAvg(); // Kobosh
 }
diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index e5947f299937a8100fd8c6b1456da9f9003c6b53..89da1f559b3d1495cea303724910a1aed360bd33 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -4020,4 +4020,6 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     default ObjectContents asObjectContents() {
         return this.getPlayerProfile().asObjectContents();
     }
+
+    long getNearbyChunkHot(); // Kobosh
 }
