From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: koboshchan <lucaszhang1001@gmail.com>
Date: Mon, 18 May 2026 05:12:26 -0700
Subject: [PATCH] Add config file for kobosh


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index db1440b09e620ce2aa9441e92b43b0d1ab8ee7a8..3e3ac4fe767855ac54989230c561624ed0eb6d51 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1024,6 +1024,7 @@ public final class CraftServer implements Server {
         }
 
         org.spigotmc.SpigotConfig.init((File) this.console.options.valueOf("spigot-settings")); // Spigot
+        org.kobosh.koboshfolia.KoboshConfig.init((File) this.console.options.valueOf("kobosh-settings")); // Kobosh
         this.console.paperConfigurations.reloadConfigs(this.console);
         for (ServerLevel world : this.console.getAllLevels()) {
             // world.serverLevelData.setDifficulty(config.difficulty); // Paper - per level difficulty
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 9d4d97fcb0f3cc94e9a263c61cc0e7e58fd322c3..1e615ddfe7db8cc421be7adfd57ef41a8e3917c4 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -157,6 +157,12 @@
                         .defaultsTo(new File("paper.yml"))
                         .describedAs("Yml file");
 
+                this.acceptsAll(asList("kobosh", "kobosh-settings"), "File for Kobosh settings")
+                        .withRequiredArg()
+                        .ofType(File.class)
+                        .defaultsTo(new File("kobosh.yml"))
+                        .describedAs("Yml file");
+
                 this.acceptsAll(asList("add-plugin", "add-extra-plugin-jar"), "Specify paths to extra plugin jars to be loaded in addition to those in the plugins folder. This argument can be specified multiple times, once for each extra plugin jar path.")
                         .withRequiredArg()
                         .ofType(File.class)
