From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: koboshchan <lucaszhang1001@gmail.com>
Date: Tue, 19 May 2026 03:59:28 -0700
Subject: [PATCH] org.kobosh -> com.kobosh


diff --git a/src/main/java/org/kobosh/koboshfolia/KoboshConfig.java b/src/main/java/com/kobosh/koboshfolia/KoboshConfig.java
similarity index 99%
rename from src/main/java/org/kobosh/koboshfolia/KoboshConfig.java
rename to src/main/java/com/kobosh/koboshfolia/KoboshConfig.java
index c8ac26e755c3596df0b8f22c523ba6fe9d427a10..a94b434900fb74b471763b8a325911c07c863f49 100644
--- a/src/main/java/org/kobosh/koboshfolia/KoboshConfig.java
+++ b/src/main/java/com/kobosh/koboshfolia/KoboshConfig.java
@@ -1,4 +1,4 @@
-package org.kobosh.koboshfolia;
+package com.kobosh.koboshfolia;
 
 import com.google.common.base.Throwables;
 import org.bukkit.Bukkit;
@@ -8,7 +8,7 @@ import org.bukkit.configuration.file.YamlConfiguration;
 import java.io.File;
 import java.io.IOException;
 import java.util.logging.Level;
-import org.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
+import com.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
 
 public class KoboshConfig {
     private static final String HEADER = "This is the main configuration file for KoboshFolia.\n";
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/AbstractGlobalServerBar.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/AbstractGlobalServerBar.java
similarity index 98%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/AbstractGlobalServerBar.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/AbstractGlobalServerBar.java
index 6965ebbacab5f3050c451d68daaf4548b5971e96..43e9949cd6ff175cec6bc8d286627ba5d2fa87a1 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/AbstractGlobalServerBar.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/AbstractGlobalServerBar.java
@@ -1,4 +1,4 @@
-package org.kobosh.koboshfolia.tpsbar;
+package com.kobosh.koboshfolia.tpsbar;
 
 import com.google.common.collect.Maps;
 import com.mojang.logging.LogUtils;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/EnumBarType.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/EnumBarType.java
similarity index 88%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/EnumBarType.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/EnumBarType.java
index ce7ee84a7ffc2c950086a90d4323ad1b2f0e583a..f8aef5964f115219ce924402722295d3a297fb67 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/EnumBarType.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/EnumBarType.java
@@ -1,10 +1,10 @@
-package org.kobosh.koboshfolia.tpsbar;
+package com.kobosh.koboshfolia.tpsbar;
 
 import com.mojang.datafixers.util.Pair;
-import org.kobosh.koboshfolia.tpsbar.AbstractGlobalServerBar;
-import org.kobosh.koboshfolia.tpsbar.GlobalServerMemoryBar;
-import org.kobosh.koboshfolia.tpsbar.GlobalServerRegionBar;
-import org.kobosh.koboshfolia.tpsbar.GlobalServerTpsBar;
+import com.kobosh.koboshfolia.tpsbar.AbstractGlobalServerBar;
+import com.kobosh.koboshfolia.tpsbar.GlobalServerMemoryBar;
+import com.kobosh.koboshfolia.tpsbar.GlobalServerRegionBar;
+import com.kobosh.koboshfolia.tpsbar.GlobalServerTpsBar;
 import org.jetbrains.annotations.NotNull;
 
 public enum EnumBarType {
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/EnumStatusBarDisplay.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/EnumStatusBarDisplay.java
similarity index 67%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/EnumStatusBarDisplay.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/EnumStatusBarDisplay.java
index 5da6be9afcf8e0f098b55ad95dcb6b33a3cbbb81..86c4d6772965ce316acd02f2dc1f2772c6690d23 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/EnumStatusBarDisplay.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/EnumStatusBarDisplay.java
@@ -1,4 +1,4 @@
-package org.kobosh.koboshfolia.tpsbar;
+package com.kobosh.koboshfolia.tpsbar;
 
 public enum EnumStatusBarDisplay {
     BOSS_BAR,
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerBarManager.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerBarManager.java
similarity index 84%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerBarManager.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerBarManager.java
index 4d6b89fd4dbdf197efd61bb049e2c833420594e3..fcdd69d0f74ef56e4a31790dcb51e2279b65c8c5 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerBarManager.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerBarManager.java
@@ -1,6 +1,6 @@
-package org.kobosh.koboshfolia.tpsbar;
+package com.kobosh.koboshfolia.tpsbar;
 
-import org.kobosh.koboshfolia.tpsbar.EnumBarType;
+import com.kobosh.koboshfolia.tpsbar.EnumBarType;
 
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerMemoryBar.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerMemoryBar.java
similarity index 96%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerMemoryBar.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerMemoryBar.java
index 781720136846c2fbfe5e3c34f2aad7c913600e6a..ae40922c866454ef2d58f1a4bac01a3f54602613 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerMemoryBar.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerMemoryBar.java
@@ -1,8 +1,8 @@
-package org.kobosh.koboshfolia.tpsbar;
+package com.kobosh.koboshfolia.tpsbar;
 
 import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
-import org.kobosh.koboshfolia.KoboshConfig;
-import org.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
+import com.kobosh.koboshfolia.KoboshConfig;
+import com.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
 import net.kyori.adventure.bossbar.BossBar;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.minimessage.MiniMessage;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerRegionBar.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerRegionBar.java
similarity index 97%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerRegionBar.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerRegionBar.java
index ad3d548308890efec5ef02a3a1831e6c0f37c249..dced95c82cfdaf4d063c5e9c6794f2aab69c134e 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerRegionBar.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerRegionBar.java
@@ -1,12 +1,12 @@
-package org.kobosh.koboshfolia.tpsbar;
+package com.kobosh.koboshfolia.tpsbar;
 
 import ca.spottedleaf.moonrise.common.time.TickData;
 import io.papermc.paper.threadedregions.ThreadedRegionizer;
 import io.papermc.paper.threadedregions.TickRegionScheduler;
 import io.papermc.paper.threadedregions.TickRegions;
 import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
-import org.kobosh.koboshfolia.KoboshConfig;
-import org.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
+import com.kobosh.koboshfolia.KoboshConfig;
+import com.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
 import net.kyori.adventure.bossbar.BossBar;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.minimessage.MiniMessage;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerTpsBar.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerTpsBar.java
similarity index 98%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerTpsBar.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerTpsBar.java
index cd397163320db94aecfbcaa6a132104d573ef1dc..ffc27673c3eb78b3242d715734b03d47a06b7548 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/GlobalServerTpsBar.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/GlobalServerTpsBar.java
@@ -1,12 +1,12 @@
-package org.kobosh.koboshfolia.tpsbar;
+package com.kobosh.koboshfolia.tpsbar;
 
 import ca.spottedleaf.moonrise.common.time.TickData;
 import io.papermc.paper.threadedregions.ThreadedRegionizer;
 import io.papermc.paper.threadedregions.TickRegionScheduler;
 import io.papermc.paper.threadedregions.TickRegions;
 import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
-import org.kobosh.koboshfolia.KoboshConfig;
-import org.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
+import com.kobosh.koboshfolia.KoboshConfig;
+import com.kobosh.koboshfolia.tpsbar.EnumStatusBarDisplay;
 import net.kyori.adventure.bossbar.BossBar;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.minimessage.MiniMessage;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/BarCommand.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/BarCommand.java
similarity index 90%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/commands/BarCommand.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/commands/BarCommand.java
index ac4c2cd512ae98e80a215722d470855aa2390e8f..9e3a3509f754e9d28ddf65348e3febd9383c5475 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/BarCommand.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/BarCommand.java
@@ -1,7 +1,7 @@
-package org.kobosh.koboshfolia.tpsbar.commands;
+package com.kobosh.koboshfolia.tpsbar.commands;
 
-import org.kobosh.koboshfolia.tpsbar.commands.sub.ToggleCommand;
-import org.kobosh.koboshfolia.tpsbar.EnumBarType;
+import com.kobosh.koboshfolia.tpsbar.commands.sub.ToggleCommand;
+import com.kobosh.koboshfolia.tpsbar.EnumBarType;
 import org.bukkit.command.CommandSender;
 import org.jetbrains.annotations.NotNull;
 import org.leavesmc.leaves.command.RootNode;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/BarSubcommand.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/BarSubcommand.java
similarity index 79%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/commands/BarSubcommand.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/commands/BarSubcommand.java
index 33c64d9da1a70fa64c9b520460a10c265c41a94a..754bdc6abb04ad4f40d06d427b1eda74b0e2c629 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/BarSubcommand.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/BarSubcommand.java
@@ -1,9 +1,9 @@
-package org.kobosh.koboshfolia.tpsbar.commands;
+package com.kobosh.koboshfolia.tpsbar.commands;
 
 import io.papermc.paper.command.brigadier.CommandSourceStack;
-import org.kobosh.koboshfolia.tpsbar.commands.sub.ConfigEditCommand;
-import org.kobosh.koboshfolia.tpsbar.commands.sub.ToggleCommand;
-import org.kobosh.koboshfolia.tpsbar.EnumBarType;
+import com.kobosh.koboshfolia.tpsbar.commands.sub.ConfigEditCommand;
+import com.kobosh.koboshfolia.tpsbar.commands.sub.ToggleCommand;
+import com.kobosh.koboshfolia.tpsbar.EnumBarType;
 import org.bukkit.command.CommandSender;
 import org.jetbrains.annotations.NotNull;
 import org.leavesmc.leaves.command.CommandNode;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/sub/ConfigEditCommand.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/sub/ConfigEditCommand.java
similarity index 91%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/commands/sub/ConfigEditCommand.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/commands/sub/ConfigEditCommand.java
index 6657154bc8d51fb0cab2ca4cea347c9993af85de..9bcc9ae462f6204e856d246b12cdbb0072a5152b 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/sub/ConfigEditCommand.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/sub/ConfigEditCommand.java
@@ -1,11 +1,11 @@
-package org.kobosh.koboshfolia.tpsbar.commands.sub;
+package com.kobosh.koboshfolia.tpsbar.commands.sub;
 
 import com.mojang.brigadier.arguments.BoolArgumentType;
 import me.earthme.luminol.config.ConfigManager;
 import me.earthme.luminol.config.ConfigsInstance;
-import org.kobosh.koboshfolia.tpsbar.EnumBarType;
-import org.kobosh.koboshfolia.tpsbar.AbstractGlobalServerBar;
-import org.kobosh.koboshfolia.tpsbar.GlobalServerBarManager;
+import com.kobosh.koboshfolia.tpsbar.EnumBarType;
+import com.kobosh.koboshfolia.tpsbar.AbstractGlobalServerBar;
+import com.kobosh.koboshfolia.tpsbar.GlobalServerBarManager;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.format.TextColor;
 import org.jetbrains.annotations.NotNull;
diff --git a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/sub/ToggleCommand.java b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/sub/ToggleCommand.java
similarity index 94%
rename from src/main/java/org/kobosh/koboshfolia/tpsbar/commands/sub/ToggleCommand.java
rename to src/main/java/com/kobosh/koboshfolia/tpsbar/commands/sub/ToggleCommand.java
index fc26b94d240a740b4fbdb0927bea4840f93e891a..e7ddd522666aad0304c50fa22751f05e38b1b121 100644
--- a/src/main/java/org/kobosh/koboshfolia/tpsbar/commands/sub/ToggleCommand.java
+++ b/src/main/java/com/kobosh/koboshfolia/tpsbar/commands/sub/ToggleCommand.java
@@ -1,4 +1,4 @@
-package org.kobosh.koboshfolia.tpsbar.commands.sub;
+package com.kobosh.koboshfolia.tpsbar.commands.sub;
 
 import com.mojang.brigadier.arguments.StringArgumentType;
 import com.mojang.brigadier.builder.ArgumentBuilder;
@@ -9,10 +9,10 @@ import com.mojang.brigadier.suggestion.SuggestionsBuilder;
 import io.papermc.paper.command.brigadier.CommandSourceStack;
 import io.papermc.paper.command.brigadier.Commands;
 import io.papermc.paper.command.brigadier.PaperCommands;
-import org.kobosh.koboshfolia.tpsbar.commands.BarCommand;
-import org.kobosh.koboshfolia.tpsbar.EnumBarType;
-import org.kobosh.koboshfolia.tpsbar.AbstractGlobalServerBar;
-import org.kobosh.koboshfolia.tpsbar.GlobalServerBarManager;
+import com.kobosh.koboshfolia.tpsbar.commands.BarCommand;
+import com.kobosh.koboshfolia.tpsbar.EnumBarType;
+import com.kobosh.koboshfolia.tpsbar.AbstractGlobalServerBar;
+import com.kobosh.koboshfolia.tpsbar.GlobalServerBarManager;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.format.TextColor;
 import org.bukkit.Bukkit;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 54c37bdf8384c6424b391d99d1359d032d2a2ba1..01378a039e34eb9b27735dc8df00baf687fd1891 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1017,7 +1017,7 @@ public final class CraftServer implements Server {
         }
 
         org.spigotmc.SpigotConfig.init((File) this.console.options.valueOf("spigot-settings")); // Spigot
-        org.kobosh.koboshfolia.KoboshConfig.init((File) this.console.options.valueOf("kobosh-settings")); // Kobosh
+        com.kobosh.koboshfolia.KoboshConfig.init((File) this.console.options.valueOf("kobosh-settings")); // Kobosh
         this.console.paperConfigurations.reloadConfigs(this.console);
         for (ServerLevel world : this.console.getAllLevels()) {
             // world.serverLevelData.setDifficulty(config.difficulty); // Paper - per level difficulty
