From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: koboshchan <lucaszhang1001@gmail.com>
Date: Tue, 19 May 2026 09:04:10 -0700
Subject: [PATCH] Leaf: simpler vanilla shapeless comparison

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftShapelessRecipe.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftShapelessRecipe.java
index 6a7bb225f044c9f299d6a3e703de82a1093878b4..a637c3d80d0af42842ef36d49bbdf67858e9655a 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftShapelessRecipe.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftShapelessRecipe.java
@@ -45,12 +45,12 @@ public class CraftShapelessRecipe extends ShapelessRecipe implements CraftRecipe
         for (org.bukkit.inventory.RecipeChoice choice : choices) {
             ingredients.add(CraftRecipe.toIngredient(choice, true));
         }
-
         net.minecraft.world.item.crafting.ShapelessRecipe recipe = new net.minecraft.world.item.crafting.ShapelessRecipe(
             new net.minecraft.world.item.crafting.Recipe.CommonInfo(true),
             new net.minecraft.world.item.crafting.CraftingRecipe.CraftingBookInfo(CraftRecipe.getCategory(this.getCategory()), this.getGroup()),
             CraftItemStack.asTemplate(this.getResult()),
-            ingredients
+            ingredients,
+            true // Kobosh - Leaf - mark Bukkit shapeless recipes
         );
         MinecraftServer.getServer().getRecipeManager().addRecipe(new RecipeHolder<>(CraftNamespacedKey.toResourceKey(Registries.RECIPE, this.getKey()), recipe));
     }
