--- a/net/minecraft/core/Direction.java
+++ b/net/minecraft/core/Direction.java
@@ -223,7 +_,7 @@
     }
 
     public Direction getOpposite() {
-        return this.opposite; // Paper - optimise collisions
+        return VALUES[this.oppositeIndex]; // Kobosh - Lithium fastutil
     }
 
     public Direction getClockWise(final Direction.Axis axis) {
@@ -355,7 +_,7 @@
     }
 
     public static Direction getRandom(final RandomSource random) {
-        return Util.getRandom(VALUES, random);
+        return VALUES[random.nextInt(VALUES.length)]; // Kobosh - Lithium fastutil
     }
 
     public static Direction getApproximateNearest(final double dx, final double dy, final double dz) {
