--- a/net/minecraft/world/level/levelgen/synth/PerlinNoise.java
+++ b/net/minecraft/world/level/levelgen/synth/PerlinNoise.java
@@ -26,6 +_,10 @@
     private final double lowestFreqValueFactor;
     private final double lowestFreqInputFactor;
     private final double maxValue;
+    // Kobosh start - Gale - optimize noise generation
+    private final int octaveSamplersCount;
+    private final double[] amplitudesArray;
+    // Kobosh end - Gale - optimize noise generation
 
     @Deprecated
     public static PerlinNoise createLegacyForBlendedNoise(RandomSource random, IntStream octaves) {
@@ -127,6 +_,10 @@
         this.lowestFreqInputFactor = Math.pow(2.0, -i);
         this.lowestFreqValueFactor = Math.pow(2.0, size - 1) / (Math.pow(2.0, size) - 1.0);
         this.maxValue = this.edgeValue(2.0);
+        // Kobosh start - Gale - optimize noise generation
+        this.octaveSamplersCount = this.noiseLevels.length;
+        this.amplitudesArray = this.amplitudes.toDoubleArray();
+        // Kobosh end - Gale - optimize noise generation
     }
 
     protected double maxValue() {
@@ -138,7 +_,23 @@
     }
 
     public double getValue(double x, double y, double z) {
-        return this.getValue(x, y, z, 0.0, 0.0, false);
+        // Kobosh start - Gale - optimize noise generation
+        double d = 0.0;
+        double e = this.lowestFreqInputFactor;
+        double f = this.lowestFreqValueFactor;
+
+        for (int i = 0; i < this.octaveSamplersCount; ++i) {
+            ImprovedNoise improvedNoise = this.noiseLevels[i];
+            if (improvedNoise != null) {
+                @SuppressWarnings("deprecation")
+                double g = improvedNoise.noise(wrap(x * e), wrap(y * e), wrap(z * e), 0.0, 0.0);
+                d += this.amplitudesArray[i] * g * f;
+            }
+            e *= 2.0;
+            f /= 2.0;
+        }
+        return d;
+        // Kobosh end - Gale - optimize noise generation
     }
 
     @Deprecated
