--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -180,6 +_,7 @@
 import net.minecraft.world.level.block.entity.CommandBlockEntity;
 import net.minecraft.world.level.block.entity.SignBlockEntity;
 import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.chunk.LevelChunk;
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.level.gamerules.GameRules;
 import net.minecraft.world.level.portal.TeleportTransition;
@@ -240,6 +_,7 @@
     private int lastSentFood = -99999999;
     private boolean lastFoodSaturationZero = true;
     public int lastSentExp = -99999999;
+    private volatile long nearbyChunkHot = 0L;
     private ChatVisiblity chatVisibility = ChatVisiblity.FULL;
     public ParticleStatus particleStatus = ParticleStatus.ALL;
     private boolean canChatColor = true;
@@ -1050,6 +_,7 @@
 
             if (this.tickCount % 20 == 0) {
                 CriteriaTriggers.LOCATION.trigger(this);
+                this.nearbyChunkHot = this.refreshNearbyChunkHot();
             }
 
             // CraftBukkit start - initialize oldLevel, fire PlayerLevelChangeEvent, and tick client-sided world border
@@ -1081,6 +_,24 @@
                 this.connection.send(updatePacket);
             }
         }
+    }
+
+    public long getNearbyChunkHot() {
+        return this.nearbyChunkHot;
+    }
+
+    private long refreshNearbyChunkHot() {
+        long total = 0L;
+        int searchRadius = ((ServerLevel)this.level()).moonrise$getViewDistanceHolder().getViewDistances().tickViewDistance();
+        for (int chunkX = this.moonrise$getSectionX() - searchRadius; chunkX <= this.moonrise$getSectionX() + searchRadius; ++chunkX) {
+            for (int chunkZ = this.moonrise$getSectionZ() - searchRadius; chunkZ <= this.moonrise$getSectionZ() + searchRadius; ++chunkZ) {
+                LevelChunk targetChunk = this.level().getChunkIfLoaded(chunkX, chunkZ);
+                if (targetChunk != null) {
+                    total += targetChunk.getChunkHot().getAverage();
+                }
+            }
+        }
+        return total;
     }
 
     @Override
